.github/workflows/tests.yaml

```diff
@@ -0,0 +1,75 @@
+name: Tests and linting
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron:  '0 3 * * *'
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-version: ['7.2', '7.3', '7.4']
+        dependencies: ['']
+        include:
+          - { php-version: '7.2', dependencies: '--prefer-lowest --prefer-stable' }
+
+    name: PHP ${{ matrix.php-version }} ${{ matrix.dependencies }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          extensions: mbstring, intl, zip
+          coverage: xdebug
+          tools: composer:v1 # TODO update to v2
+
+      - name: Install dependencies
+        run: composer update --no-progress ${{ matrix.dependencies }}
+
+      - name: Run tests
+        run: vendor/bin/phpunit --coverage-clover coverage-clover.xml
+
+      - name: Submit coverage to Coveralls
+        env:
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          composer global require --no-progress --dev php-coveralls/php-coveralls guzzlehttp/guzzle:^6.5
+          ~/.composer/vendor/bin/php-coveralls -v
+
+  codestyle:
+      name: "Code style and static analysis"
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+
+        - name: Setup PHP
+          uses: shivammathur/setup-php@v2
+          with:
+            php-version: '7.4'
+            extensions: mbstring, intl
+            tools: composer:v2 # TODO update to v2
```

Comment:
I dont think, this is correct - it should either be `:v1` or without a todo comment